### PR TITLE
random: use arc4random on OpenBSD

### PR DESCRIPTION
**Pull request by theStack**
Inspired by a discussion on obtaining randomness on various OSes in a secp256k1 PR (https://github.com/bitcoin-core/secp256k1/pull/748#discussion_r524605472, see also https://bitcoincore.reviews/libsecp256k1-748), I think it makes sense to follow best practices and use `arc4random_buf` rather than `getentropy` on OpenBSD in our random module.

The [getentropy(2) man page](https://man.openbsd.org/getentropy.2) states:
```
getentropy() is not intended for regular code; please use the
arc4random(3) family of functions instead.
```

The [arc4random(3) man page](https://man.openbsd.org/arc4random.3) states:

```
Use of these functions is encouraged for almost all random number
consumption because the other interfaces are deficient in either quality,
portability, standardization, or availability.
```
On the linked PR discussion worries about using RC4 internally has been expressed (see https://security.stackexchange.com/questions/85601/is-arc4random-secure-enough/172905#172905), but this would only affect users of OpenBSD <5.5, using a version that was released more than 8 years ago.

### CODE REVIEW

File: src/random.cpp [hunk 1]
```diff
@@ -305,16 +305,14 @@ void GetOSRand(unsigned char *ent32)
             RandFailure();
         }
     }
-#elif defined(HAVE_GETENTROPY) && defined(__OpenBSD__)
```
Comment (MarcoFalke) [inline src/random.cpp:308]: nit: I think you can remove `HAVE_GETENTROPY` now?

Comment (theStack) [reply src/random.cpp:308]: Good idea, done.

**Comment by theStack** [comment]
Added another commit that removed the now unused getentropy detection (HAVE_GETENTROPY), [as suggested by MarcoFalke](https://github.com/bitcoin/bitcoin/pull/24238#discussion_r797776339). To my understanding, the inclusion of `<unistd.h>` within the pre-processor condition "`defined(HAVE_GETENTROPY_RAND) && defined(MAC_OSX)`" could also be removed, but I'm not completely sure and don't have that OS available to test. Maybe a MacOS user could comment on this.

**Comment by kristapsk** [comment]
Concept ACK

**Comment by laanwj** [comment]
Concept ACK. I'm surprised `getentropy` was only used on OpenBSD. Clearly, if we use OS-specific functions on other OSes, and already have a OpenBSD specific path, then it also makes sense to use the suggested function.

**Comment by gruve-p** [comment]
Concept ACK

**Comment by laanwj** [comment]
Tested ACK 0c49e52b22be1baa8d51670e4f3c437fd3c0baa7
I verified that `arc4random_buf (buf=?, n=32)` is being called on OpenBSD 7.0, and the buffer is filled with different random bytes every time.

**Pull request merged**